File: app/channels/counter_channel.rb
class CounterChannel < ApplicationCable::Channel
  def subscribed
    # pubsub
    stream_from "counter_channel"
  end

  def receiving(data)
    # the value of the counter is incremented client-side
    counter = Counter.create!(nb: data['countPG'] )
    msg = {}
    msg['countPG'] = data['countPG']
    puts msg
    ActionCable.server.broadcast('counter_channel', msg.as_json) if counter.valid?
  end

  def unsubscribed
    stop_all_streams
  end
end

File: app/javascript/channels/counter_channel.js
import consumer from "./consumer";

const counterChannel = consumer.subscriptions.create(
  { channel: "CounterChannel" },
  {
    connected() {
      console.log("click connected");
    },
    disconnected() {},
    received: function (data) {
      // overwritten in Button.js
    },
    sending(data) {
      this.perform("receiving", { countPG: data });
    },
  }
);
export default counterChannel;
